ovos_plugin_manager/templates/tts.py

```diff
@@ -238,111 +144,14 @@ def runtime_requirements(self):
-        """Load phonetic spellings of words as dictionary."""
-        path = join('text', self.lang.lower(), 'phonetic_spellings.txt')
-        try:
-            spellings_file = resolve_resource_file(path, config=config or Configuration())
-        except:
-            LOG.debug('Failed to locate phonetic spellings resouce file.')
-            return {}
-        if not spellings_file:
-            return {}
-        try:
-            with open(spellings_file) as f:
-                lines = filter(bool, f.read().split('\n'))
-            lines = [i.split(':') for i in lines]
-            return {key.strip(): value.strip() for key, value in lines}
-        except ValueError:
-            LOG.exception('Failed to load phonetic spellings.')
-            return {}
-
-    def begin_audio(self):
-        """Helper function for child classes to call in execute()"""
-        self.add_metric({"metric_type": "tts.start"})
-
-    def end_audio(self, listen=False):
-        """Helper cleanup function for child classes to call in execute().
-
-        Arguments:
-            listen (bool): DEPRECATED: indication if listening trigger should be sent.
-        """
-        self.add_metric({"metric_type": "tts.end"})
-        self.stopwatch.stop()
-
-    def init(self, bus=None, playback=None):
-        """ Performs intial setup of TTS object.
-
-        Arguments:
-            bus:    OpenVoiceOS messagebus connection
+    def available_languages(self) -> set:
+        """Return languages supported by this TTS implementation in this state
+        This property should be overridden by the derived class to advertise
+        what languages that engine supports.
+        Returns:
+            set: supported languages
         """
-        self.bus = bus or BUS()
-        if playback is None:
-            LOG.warning("PlaybackThread should be inited by ovos-audio, initing via plugin has been deprecated, "
-                        "please pass playback=PlaybackThread() to TTS.init")
-            if TTS.playback:
-                playback.shutdown()
-            playback = PlaybackThread(TTS.queue, self.bus)  # compat
-            playback.start()
-        self._init_playback(playback)
-        self.add_metric({"metric_type": "tts.setup"})
-
-    def _init_playback(self, playback):
-        TTS.playback = playback
-        TTS.playback.set_bus(self.bus)
-        TTS.playback.attach_tts(self)
-        if not TTS.playback.enclosure:
-            TTS.playback.enclosure = EnclosureAPI(self.bus)
-
-        if not TTS.playback.is_running:
-            TTS.playback.start()
-
-    @property
-    def enclosure(self):
-        if not TTS.playback.enclosure:
-            bus = TTS.playback.bus or self.bus
-            TTS.playback.enclosure = EnclosureAPI(bus)
-        return TTS.playback.enclosure
-
-    @enclosure.setter
-    def enclosure(self, val):
-        TTS.playback.enclosure = val
+        return set()
 
     def get_tts(self, sentence, wav_file, lang=None):
```

Comment:
```suggestion
    def get_tts(self, sentence, wav_file, lang=None) -> Tuple[str, Optional[str]]:
```

Reply:
I could see the return value on this introducing a lot of bugs in plugins unless it's made explicit

Reply:
this return value is the default method all plugins override, we cant really change it without breaking everything
it has been this way since mycroft day 1

Reply:
Not asking to change the actual return value, but rather make it explicit what is being returned

Reply:
i dont oppose that, but adding typing should be it's own PR and target all plugins